spog/api/src/endpoints/suggestion.rs

```diff
@@ -0,0 +1,88 @@
+use crate::error::Error;
+use crate::search::QueryParams;
+use crate::service::v11y::V11yService;
+use actix_web::web::ServiceConfig;
+use actix_web::{web, Responder};
+use cvss::v3::Score;
+use spog_model::prelude::*;
+use std::fmt::Write;
+use std::sync::Arc;
+use trustification_auth::authenticator::Authenticator;
+use trustification_infrastructure::new_auth;
+
+pub(crate) fn configure(auth: Option<Arc<Authenticator>>) -> impl FnOnce(&mut ServiceConfig) {
+    |config: &mut ServiceConfig| {
+        config.service(
+            web::resource("/api/v1/suggestions/search")
+                .wrap(new_auth!(auth))
+                .to(search_suggestions),
+        );
+    }
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+struct SearchQuery {
+    #[serde(default)]
+    term: String,
+}
+
+#[utoipa::path(
+    get,
+    path = "/api/v1/suggestions/search",
+        responses(
+            (status = OK, description = "The resulting suggestions", body = Vec<Suggestion>),
+        ),
+    params(
+        ("term" = String, Query, description = "The search term to provide suggestions for")
+    )
+)]
+async fn search_suggestions(
+    v11y: web::Data<V11yService>,
+    query: web::Query<SearchQuery>,
+) -> Result<impl Responder, Error> {
+    Ok(web::Json(create_suggestions(&v11y, &query.term).await?))
+}
+
+async fn create_suggestions(v11y: &V11yService, term: &str) -> Result<Vec<Suggestion>, Error> {
+    let term = term.replace("\"", "");
+    let q = format!(r#""{term}" is:published"#);
+
+    let result = v11y.search(QueryParams { q, offset: 0, limit: 3 }).await?;
```

Comment:
Currently the "suggestions" search is applied only to "packages (v11y)". I just want to make sure we can extend the "suggestions" also to "sboms" and "packages". The limitation I see (i might be wrong) is that each entity "cve", "sbom", and "packages" are not related and don't share a single index where we can query the "suggestions".

- The final objective might have a single search endpoint and get suggestions for all entities "cve", "sbom", and "packages". How easy or difficult it might be to extend the current code to include those 3 entities in the future? I'm not saying we have to do it now, we can start small and incrementally enhance things.

Reply:
Absolutely. The one of the next steps I want to take a look at is to add at least SBOMs. Advisories might be a bit trickier. Not due to technical difficulties, but due to aggregating the results. Otherwise we might have two CVEs with the same ID (one from raw CVE, the other from a VEX with the same ID). So it's more about defining what we want.